sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -149,10 +149,24 @@ case class AggregateExpression(
       case PartialMerge => "merge_"
       case Final | Complete => ""
     }
-    prefix + aggregateFunction.toAggString(isDistinct)
+    val aggFuncStr = prefix + aggregateFunction.toAggString(isDistinct)
+    mode match {
+      case Partial | Complete if filter.isDefined =>
```

Comment:
Although we made filter evaluated in first Aggregate.
I think there should just judge that the filter is defined.

Reply:
Doing so shows a filter in a second aggregate in physical plans like this;
```
== Physical Plan ==
HashAggregate(keys=[k#0], functions=[sum(cast(v#1 as bigint)) filter ((v#1 > 3))], output=[k#0, sum(v) FILTER ((v > 3))#44L])
+- Exchange hashpartitioning(k#0, 200), true, [id=#154]
   +- HashAggregate(keys=[k#0], functions=[partial_sum(cast(v#1 as bigint)) filter ((v#1 > 3))], output=[k#0, sum#48L])
      +- *(1) ColumnarToRow
         +- FileScan parquet default.t[k#0,v#1] Batched: true, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex[file:/Users/maropu/Repositories/spark/spark-master/spark-warehouse/t], PartitionFilters: [], PushedFilters: [], ReadSchema: struct<k:int,v:int>
```

Reply:
I mean the mode is `PartialMerge`.
If we only need to show filter in physical plans after rewrite, this is OK.